packages/core/src/EntityManager.ts

```diff
@@ -488,6 +492,8 @@ export class EntityManager<D extends IDatabaseDriver = IDatabaseDriver> {
     Entity extends object,
     Hint extends string = never,
   >(entityName: EntityName<Entity>, where: FilterQuery<Entity>, options: FindOneOptions<Entity, Hint> = {}): Promise<Loaded<Entity, Hint> | null> {
+    options.schema ??= this._schema;
+
```

Comment:
```suggestion
    Entity extends object,
    Hint extends string = never,
  >(entityName: EntityName<Entity>, where: FilterQuery<Entity>, options: FindOneOptions<Entity, Hint> = {}): Promise<Loaded<Entity, Hint> | null> {
    options.schema ??= this.schema;

```
If you use NestJs package and get a repo for example
```
constructor(@InjectRepository(Comment) private readonly commentRepo: EntityRepository<Comment>) {}

findComment() {
        const repoManager = this.commentRepo.getEntityManager();
        // Will set schema on the contextual EntityManager
        repoManager.schema = 'client_4';
        
       // client_4
       // Will output the contextual schema
        console.log(repoManager.schema)

       // undefined
       // Will output the schema on global entity manager
        console.log(repoManager._schema)

        // Will query against the orm schema since we use this._schema in findOne
        const comment = await this.commentRepo.findOne({
            id: 2,
        });
}
```

So I guess we need to change to use the contextual schema (`this.schema`)

Reply:
Right, what you should do is to use the EM from context, all the methods first get that and work with it instead of using `this`.

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/EntityManager.ts#L500

No need to duplicate the `getContext` calls for this.

Reply:
Great, that should solve the problem

Reply:
I will try and add a test for context schema as well